src/core/main.c

```diff
@@ -1753,6 +1756,27 @@ static void update_cpu_affinity(bool skip_setup) {
                 log_warning_errno(errno, "Failed to set CPU affinity: %m");
 }
 
+static void update_numa_policy(bool skip_setup) {
+        int r;
+        _cleanup_free_ char *nodes = NULL;
+        const char * policy = NULL;
```

Comment:
these two are no used outside of the `if (DEBUG_LOGGING)`, please move them into that scope hence. (usually we try to define variables at minimal scope only, except for `int r`)